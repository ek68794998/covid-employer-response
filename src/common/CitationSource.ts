export class CitationSource {
	public date?: string;

	public link: string = "";

	public source: string = "";

	public title: string = "";
}
