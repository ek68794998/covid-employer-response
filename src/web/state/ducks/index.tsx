import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});