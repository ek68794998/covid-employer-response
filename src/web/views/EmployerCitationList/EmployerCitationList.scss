ul.employer-citations {
	font-size: 14px;
	padding: 0;

	li {
		list-style: none;
		margin: 8px 0;
	}
}

h3.citation-type {
	border-bottom: 1px dotted rgba(0, 0, 0, 0.4);
	cursor: help;
	display: inline;
	font-size: 16px;
	margin: 0;
}