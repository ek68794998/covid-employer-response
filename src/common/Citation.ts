import { CitationSource } from "./CitationSource";
import { CitationType } from "./CitationType";

export class Citation {
	public positivity: number = 0;

	public sources?: CitationSource[];

	public summary: string = "";

	public type?: CitationType;
}
