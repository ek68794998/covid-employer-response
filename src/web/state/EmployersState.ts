import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}