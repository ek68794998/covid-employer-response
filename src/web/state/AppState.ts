import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}