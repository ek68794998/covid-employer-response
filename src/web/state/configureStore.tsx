import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;