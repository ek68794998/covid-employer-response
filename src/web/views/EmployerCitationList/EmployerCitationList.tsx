import React from "react";
import { useSelector } from "react-redux";
import { RouteProps } from "react-router-dom";

import { Citation } from "../../../common/Citation";
import { CitationType } from "../../../common/CitationType";
import { LocalizedStrings } from "../../../common/LocalizedStrings";

import { AppState } from "../../state/AppState";
import { getStrings } from "../../state/ducks/localization/selectors";

import EmployerCitation from "../EmployerCitation/EmployerCitation";

import "./EmployerCitationList.scss";

interface Props extends RouteProps {
	citations: Citation[];

	citationSourceBase: number;

	citationType: CitationType;
}

const EmployerCitationList: React.FC<Props> = (props: Props): React.ReactElement => {
	const strings: LocalizedStrings = useSelector((state: AppState) => getStrings(state));

	const { citations, citationSourceBase, citationType } = props;

	let citationSource: number = citationSourceBase;

	const getCitationComponent = (c: Citation, i: number): JSX.Element => {
		const component: JSX.Element =
			<li key={i}>{<EmployerCitation citation={c} citationSourceBase={citationSource} />}</li>;

		citationSource += c.sources?.length || 0;

		return component;
	};

	return (
		<div>
			<h3
				className="citation-type"
				title={strings.citationTypeDescriptions[citationType]}
			>
				{strings.citationTypes[citationType]}
			</h3>
			<ul className="employer-citations">
				{citations.map(getCitationComponent)}
			</ul>
		</div>
	);
};

export default EmployerCitationList as any;
