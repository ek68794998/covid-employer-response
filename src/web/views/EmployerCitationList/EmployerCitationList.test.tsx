import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { getPlocStringsAsync, mockComponent } from "../../../__tests__/TestUtils";
import { Citation } from "../../../common/Citation";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import EmployerCitationList from "./EmployerCitationList";

jest.mock(
	"../EmployerCitation/EmployerCitation",
	() => mockComponent("EmployerCitation"));

describe("<EmployerCitationList />", () => {
	test("renders without exploding", async () => {
		const store: Store<AppState, AnyAction> = configureStore({ strings: await getPlocStringsAsync() });

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerCitationList citations={[ new Citation() ]} />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
