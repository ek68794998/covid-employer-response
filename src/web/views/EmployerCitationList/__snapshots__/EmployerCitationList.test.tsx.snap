// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployerCitationList /> renders without exploding 1`] = `
<div>
  <h3
    className="EmployerCitationList__Header"
    title="publication desc value"
  >
    publication value
  </h3>
  <ul
    className="EmployerCitationList__Citations"
  >
    <li>
      [component: EmployerCitation citation=([object Object]) citationSourceBase=(0) /]
    </li>
  </ul>
</div>
`;
